tasks/install-package.yml

```diff
@@ -0,0 +1,114 @@
+- name: install the rpm repo's key
+  rpm_key:
+    state: present
+    key: "{{ rpm_repo_key }}"
+  when: ansible_os_family == "RedHat"
+
+- name: install apt repo keys
+  apt_key:
+    state: present
+    url: "{{ item.url }}"
+    id: "{{ item.id }}"
+  when: ansible_os_family == "Debian"
+  loop:
+  - { url: "{{ deb_repo_key }}", id: "{{ deb_repo_key_id }}" }
+  - { url: "https://deb.nodesource.com/gpgkey/nodesource.gpg.key", id: "1655A0AB68576280" }
+
+- name: install the rpm repo
+  package:
+    name: "{{ rpm_repo_url }}"
+    state: present
+    disable_gpg_check: "{{ disable_gpg_check_rpm_repo | bool }}"
+  when: ansible_os_family == "RedHat"
+
+- name: install the apt repo
+  apt:
+    deb: "{{ apt_repo_url }}"
+    state: present
+    update_cache: "{{ apt_update_cache | bool }}"
+  when: ansible_os_family == "Debian"
+
+- name: remove node 14 repo if it exists
+  file:
+    path: /etc/apt/sources.list.d/nodesource.list
```

Comment:
So if you use the previously mentioned 2.0 deb file, you won't have to remove nodejs 14 repos.

Reply:
I did have some trouble with this. I changed the repo, I'll test this a bit.

Reply:
Puppet may overwrite the file, ansible was appending to it so we had both 12 and 14. I don't know if apt recognizes several versions, or at least it didn't appear to. With both enabled I could only ever get 14, even if I specified `nodejs=12`.

Reply:
You only will have 14 if you install latest deb repo file, the 2.0 one only has nodejs 12.  I think with change to use 2.0.0 repo deb file, the issue should be solved. If it's not then there's a bug with 2.0 repo deb file.

Reply:
This is repo 2.0.0 deb repo file installs

```
root@2afc78dc9886:/# cat /etc/apt/sources.list.d/nodesource.list 
deb https://deb.nodesource.com/node_12.x focal main
```
Steps I took in fresh Ubuntu 20.04 container:

```
apt update
apt install -y wget
wget -O /tmp/release.deb https://apt.osc.edu/ondemand/latest/ondemand-release-web_2.0.0_all.deb
apt install /tmp/release.deb 
```

Reply:
yep - I removed this and updated the sed and it does appear to be working.